modules/detectron/batch_permutation_op.h

```diff
@@ -32,8 +33,37 @@ class BatchPermutationOp final : public Operator<Context> {
   USE_OPERATOR_CONTEXT_FUNCTIONS;
 
   bool RunOnDevice() override {
-    // No CPU implementation for now
-    CAFFE_NOT_IMPLEMENTED;
+    const auto& X = Input(0);
+    auto& indices = Input(1);
+    auto* Y = Output(0);
+
+    CAFFE_ENFORCE(indices.ndim() == 1, "indices must be 1-d");
```

Comment:
Please use CAFFE_ENFORCE_EQ, then when it fails the error log will be more detailed.